models/DataObject/ClassDefinition/Data/Multiselect.php

```diff
@@ -105,18 +113,46 @@ public function getRenderType(): ?string
         return $this->renderType;
     }
 
+    public function setDefaultValue(array $defaultValue): static
+    {
+        $this->defaultValue = $defaultValue;
+
+        return $this;
+    }
+
+    public function getDefaultValue(): ?array
+    {
+        return $this->defaultValue;
+    }
+
     /**
      * @see ResourcePersistenceAwareInterface::getDataForResource
      *
      *
      */
     public function getDataForResource(mixed $data, DataObject\Concrete $object = null, array $params = []): ?string
     {
-        if (is_array($data)) {
+        if (!$this->isEmpty($data) && is_array($data)) {
             return implode(',', $data);
         }
 
-        return null;
+        $defaultValue = $this->handleDefaultValue($data, $object, $params);
+
+        if (null === $defaultValue) {
+            return '';
+        }
+
+        if (is_array($defaultValue)) {
+            $defaultValue = array_map(function ($v) {
+                if(!isset($v['value'])) {
+                    return $v;
+                }
+
+                return $v['value'];
+            }, $defaultValue);
+        }
+
+        return implode(',', $defaultValue);
```

Comment:
Here we expect `$defaultValue` to be an array. If it isn't, this is throwing an exception. If it always is, we wouldn't need the condition above. But it can also be a string, I guess.

Thus, I think we should do it like this instead:

```suggestion
        if (is_array($defaultValue)) {
            return implode(',', array_map(fn ($v) => $v['value'] ?? $v, $defaultValue));
        }

        return $defaultValue ?? '';
```